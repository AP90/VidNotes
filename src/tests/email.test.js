import React from 'react';
import { shallow } from 'enzyme';

import Email from '../components/email';

describe('<Email />', () => {
    it('should render email component', () => {
        const wrapper = shallow(<Email />);
        expect(wrapper).toMatchSnapshot();
    });
});
