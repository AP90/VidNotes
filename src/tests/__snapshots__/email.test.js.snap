// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Email /> should render email component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Email />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <span
                    className="glyphicon glyphicon-envelope"
          />
           Send
</h2>,
        <p
          className="animated fadeIn email__error"
>
          Please enter a valid email address
</p>,
        <input
          className="input--shadow input--email"
          placeholder="youremail@example.com"
          type="text"
/>,
        <button
          className="btn--search"
          onClick={[Function]}
>
          <span
                    className="glyphicon glyphicon-send"
          />
</button>,
      ],
      "className": "email",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="glyphicon glyphicon-envelope"
/>,
            " Send",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "glyphicon glyphicon-envelope",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          " Send",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please enter a valid email address",
          "className": "animated fadeIn email__error",
        },
        "ref": null,
        "rendered": "Please enter a valid email address",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input--shadow input--email",
          "placeholder": "youremail@example.com",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="glyphicon glyphicon-send"
/>,
          "className": "btn--search",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "glyphicon glyphicon-send",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <span
                        className="glyphicon glyphicon-envelope"
            />
             Send
</h2>,
          <p
            className="animated fadeIn email__error"
>
            Please enter a valid email address
</p>,
          <input
            className="input--shadow input--email"
            placeholder="youremail@example.com"
            type="text"
/>,
          <button
            className="btn--search"
            onClick={[Function]}
>
            <span
                        className="glyphicon glyphicon-send"
            />
</button>,
        ],
        "className": "email",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="glyphicon glyphicon-envelope"
/>,
              " Send",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "glyphicon glyphicon-envelope",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " Send",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please enter a valid email address",
            "className": "animated fadeIn email__error",
          },
          "ref": null,
          "rendered": "Please enter a valid email address",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input--shadow input--email",
            "placeholder": "youremail@example.com",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="glyphicon glyphicon-send"
/>,
            "className": "btn--search",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "glyphicon glyphicon-send",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
